src/main/java/edu/harvard/iq/dataverse/util/DatasetFieldUtil.java

```diff
@@ -0,0 +1,42 @@
+package edu.harvard.iq.dataverse.util;
+
+import edu.harvard.iq.dataverse.DatasetField;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Toolkit for datasetFields
+ * @author alejandra.tenorio
+ * @author qqmyers
+ */
+public class DatasetFieldUtil {
+
+    private static final Logger logger = Logger.getLogger(DatasetFieldUtil.class.getCanonicalName());
+
+    /**
+     * Removed empty fields, sets field value display order.
+     * @param datasetFields the datasetFields from templates or datasetVersions we want to tidy up.
+     * @param removeBlanks if remove blank fields
+     */
+    public static void tidyUpFields(List<DatasetField> datasetFields, boolean removeBlanks) {
+        if(removeBlanks){
+            Iterator<DatasetField> dsfIt = datasetFields.iterator();
+            while (dsfIt.hasNext()) {
+                if (dsfIt.next().removeBlankDatasetFieldValues()) {
+                    dsfIt.remove();
+                }
+            }
+            Iterator<DatasetField> dsfItTrim = datasetFields.iterator();
```

Comment:
Why only trim if removeBlanks is true? If I understand, trimming would help by removing any trailing spaces in the values entered in a template, which I think is better than leaving the template alone and only trimming in datasets? This isn't a big deal if there's a reason to not trim in templates, but otherwise I'd suggest doing it all the time.

Reply:
Hi @qqmyers,
I looked inside the edu.harvard.iq.dataverse.util package and I haven't found a class where to drop the method, so I added a class.
About trimTrailingSpaces, if i don't run removeBlankDatasetFieldValues and run trimTrailingSpaces a nullpointerexception occurs in the templates, at this [line](https://github.com/CIMMYT/dataverse/blob/7874-Dataset-template-authors-ordering/src/main/java/edu/harvard/iq/dataverse/DatasetField.java#L643).

Reply:
Ah - OK. Trim can't handle null values (which is what blank fields are I assume). I guess you could add a null check there but I'm OK leaving the PR as is too - it's a minor issue since dataset metadata will always get trimmed.

Reply:
@qqmyers  thanks so much, I will keep an eye on this,  in case there are any other comments.